src/rgw/rgw_rest_user.cc

```diff
@@ -41,7 +41,14 @@ void RGWOp_User_List::execute(optional_yield y)
   std::string marker;
   RESTArgs::get_uint32(s, "max-entries", 1000, &max_entries);
   RESTArgs::get_string(s, "marker", marker, &marker);
-
+  bool exists;
+  string sval = s->info.args.get("max-entries", &exists);
+  if(!std::all_of(sval.begin(), sval.end(), ::isdigit))
```

Comment:
what is this syntax?
::isdigit

Reply:
it will check if characters in the string are digit(0-9) only from start to end

Reply:
Seen it for the first time, do you know reference for this type of syntax?

Reply:
https://cplusplus.com/reference/algorithm/all_of/

Reply:
Should be std::isdigit

Reply:
There are more than one isdigit functions (from <cctype> and <locale> headers).
https://stackoverflow.com/questions/13262444/finding-an-alphanumeric-character-in-a-string-using-find-if-and-isalnum/13262555#13262555